happi/cli.py

```diff
@@ -54,12 +54,19 @@ def happi_cli(args):
     client = happi.client.Client.from_config(cfg=args.path)
 
     if args.search:
-        if args.search[0] is 'z':
-            args.search[1] = float(args.search[1])
-        search_args = {args.search[0]: args.search[1]}
+        for i in range(len(args.search)):
+            if args.search[i] is 'z':
+                args.search[i + 1] = float(args.search[i + 1])
 
-        device = client.find_device(**search_args)
-        device.show_info()
+        search_args = {}
+        for i in range(0, len(args.search), 2):
```

Comment:
Maybe check if `len(args.search)` is an even number? Otherwise this will raise an `IndexError`